streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -577,6 +575,10 @@ class StreamingContext private[streaming] (
    * @throws IllegalStateException if the StreamingContext is already stopped.
    */
   def start(): Unit = synchronized {
+    // Registering Streaming Metrics at the start of the StreamingContext
+    assert(env != null)
+    assert(env.metricsSystem != null)
+    env.metricsSystem.registerSource(streamingSource)
```

Comment:
This metrics should be added only inside the case INITIALIZED. The `start` method should be idempotent, and the current form breaks it as calling start will incorrectly attempt to register the source twice.